src/Parsers/ASTAlterQuery.cpp

```diff
@@ -661,8 +661,7 @@ void ASTAlterQuery::formatQueryImpl(const FormatSettings & settings, FormatState
 
 void ASTAlterQuery::forEachPointerToChild(std::function<void(void**)> f)
 {
-    for (const auto & child : command_list->children)
-        child->as<ASTAlterCommand &>().forEachPointerToChild(f);
+    ASTQueryWithTableAndOutput::forEachPointerToChild(f);q
     f(reinterpret_cast<void **>(&command_list));
```

Comment:
This seems to be incorrect, because you should only iterate over your children, not over the children of your children.